subsys/bluetooth/host/att.c

```diff
@@ -161,12 +162,50 @@ static void att_req_sent(struct bt_conn *conn, void *user_data)
 	}
 }
 
+static bt_conn_tx_cb_t att_cb(struct net_buf *buf);
+
+static int att_send(struct bt_conn *conn, struct net_buf *buf,
+		    bt_conn_tx_cb_t cb, void *user_data)
+{
+	struct bt_att_hdr *hdr;
+
+	hdr = (void *)buf->data;
+
+	BT_DBG("code 0x%02x", hdr->code);
+
+	if (hdr->code == BT_ATT_OP_SIGNED_WRITE_CMD) {
+		int err;
+
+		err = bt_smp_sign(conn, buf);
+		if (err) {
+			BT_ERR("Error signing data");
+			return err;
+		}
+	}
+
+	bt_l2cap_send_cb(conn, BT_L2CAP_CID_ATT, buf, cb ? cb : att_cb(buf),
+			 user_data);
+
+	return 0;
+}
+
 static void att_pdu_sent(struct bt_conn *conn, void *user_data)
 {
 	struct bt_att *att = att_get(conn);
+	struct net_buf *buf;
 
 	BT_DBG("conn %p att %p", conn, att);
 
+	/* Send more if queued */
+	while ((buf = net_buf_get(&att->tx_queue, K_NO_WAIT))) {
+		if (buf) {
```

Comment:
This if statement looks redundant since the while-loop's condition guarantees that but will be non-NULL. The unconditional net_buf_unref at the bottom of the loop would make no sense either with this extra check.